.github/workflows/go-build-test.yml

```diff
@@ -15,25 +15,112 @@ on:
       - 'go.sum'
 
 jobs:
-  build-and-test:
+  build-and-test-amd64:
```

Comment:
Is there a reason to separate this out?

Reply:
Initially I though we have to run in a different job but it doesn't make sense. You are right, I think we can use `if: ${{ matrix.arch != 'x64' }}` and update strategy according it it. Will update it.